### ISSUE TITLE
[CI] InternalTTestTests.testReduceRandom fails

### ISSUE THREAD

**Issue by tlrx** [elastic/elasticsearch#54891]
The test InternalTTestTests.testReduceRandom failed today on CI:

```

org.elasticsearch.xpack.analytics.ttest.InternalTTestTests > testReduceRandom FAILED
    java.lang.IllegalStateException: Negative longs unsupported, use writeLong or writeZLong for negative numbers [-3299506449632365585]
        at __randomizedtesting.SeedInfo.seed([58BE989DA8C0F81:C5B64598FCD196D7]:0)
        at org.elasticsearch.common.io.stream.StreamOutput.writeVLong(StreamOutput.java:271)
        at org.elasticsearch.xpack.analytics.ttest.TTestStats.writeTo(TTestStats.java:40)
        at org.elasticsearch.xpack.analytics.ttest.UnpairedTTestState.writeTo(UnpairedTTestState.java:95)
        at org.elasticsearch.common.io.stream.StreamOutput.writeNamedWriteable(StreamOutput.java:1059)
        at org.elasticsearch.xpack.analytics.ttest.InternalTTest.doWriteTo(InternalTTest.java:43)
        at org.elasticsearch.search.aggregations.InternalAggregation.writeTo(InternalAggregation.java:209)
        at org.elasticsearch.test.ESTestCase.lambda$copyWriteable$9(ESTestCase.java:1185)
        at org.elasticsearch.test.ESTestCase.copyInstance(ESTestCase.java:1212)
        at org.elasticsearch.test.ESTestCase.copyWriteable(ESTestCase.java:1185)
        at org.elasticsearch.test.AbstractWireSerializingTestCase.copyInstance(AbstractWireSerializingTestCase.java:44)
        at org.elasticsearch.test.AbstractWireSerializingTestCase.copyInstance(AbstractWireSerializingTestCase.java:33)
        at org.elasticsearch.test.AbstractWireTestCase.copyInstance(AbstractWireTestCase.java:102)
        at org.elasticsearch.test.InternalAggregationTestCase.testReduceRandom(InternalAggregationTestCase.java:320)
```

It does not reproduce using:
```
./gradlew ':x-pack:plugin:analytics:test' --tests "org.elasticsearch.xpack.analytics.ttest.InternalTTestTests.testReduceRandom" \
  -Dtests.seed=58BE989DA8C0F81 \
  -Dtests.security.manager=true \
  -Dtests.locale=ar-YE \
  -Dtests.timezone=Brazil/East \
  -Dcompiler.java=13
```

Build scan: https://gradle-enterprise.elastic.co/s/u67n6adwithdq

**Comment by nik9000** [comment]
I've likely caused this with #54776. @imotov, would you like me to look at it or do you want it?

**Comment by imotov** [comment]
@nik9000 I assumed it's me but I didn't start looking into it, so if you have time, please go ahead.

**Comment by nik9000** [comment]
I'll pick it up now.

**Comment by nik9000** [comment]
Should be fixed by 12cf663a51f9d5201e3b2715aaf81f787056ac2c. Let me know if that causes anyone any trouble.
